vcsim parse error after revert to snapshot

```
info = <pyVmomi.VmomiSupport.LazyObject object at 0x10c44bac0>, val = 'vim.vm.Snapshot:snapshot-69'

    def CheckField(info, val):
       with _lazyLock:
          valType = Type(val)
          if val is None or (isinstance(val, list) and len(val) == 0):
             # If type of the property is an Any. We should allow this to have
             # unset items
             if not (info.flags & F_OPTIONAL) and info.type is not object:
                raise TypeError('Required field "%s" not provided (not @optional)' % info.name)
             return
          elif info.type is object:
             try:
                GetQualifiedWsdlName(valType)
                return
             except KeyError:
                raise TypeError('Unknown type for %s' % info.type.__name__)
          elif isinstance(val, info.type):
             return
          elif issubclass(info.type, list):
             # Checking the values of VMOMI array types is surprisingly complicated....
             if isinstance(val, Array):
                # 1. We've got a PyVmomi Array object, which is effectively a typed list;
                # verify that the type of the Array is a subclass of the expected type.
                if issubclass(valType.Item, info.type.Item):
                   return
                elif info.flags & F_LINK:
                   # Allow objects of expected type to be assigned to links
                   if issubclass(valType, GetVmodlType(info.expectedType)):
                      return
             elif val:
                # 2. We've got a non-empty Python list object, which is untyped;
                # walk the list and make sure that each element is a subclass
                # of the expected type.

                # Masking out F_OPTIONAL part of flags since we are checking for
                # each element of the list
                flags = info.flags & (F_LINKABLE | F_LINK)
                if flags & F_LINK:
                   if info.expectedType.endswith('[]'):
                      expectedType = info.expectedType[:-2]
                   else:
                      expectedType = info.expectedType
                   itemInfo = Object(type=info.type.Item, name=info.name, flags=flags,
                                     expectedType=expectedType)
                else:
                   itemInfo = Object(type=info.type.Item, name=info.name, flags=flags)
                for it in val:
                   CheckField(itemInfo, it)
                return
             else:
                # 3. We've got None or an empty Python list object;
                # no checking required, since the result will be an empty array.
                return
          elif info.type is type and valType is type(Exception) \
            or issubclass(info.type, int) and issubclass(valType, int) \
            or issubclass(info.type, long) and (issubclass(valType, int) or \
                                                issubclass(valType, long)) \
            or issubclass(info.type, float) and issubclass(valType, float) \
            or issubclass(info.type, string_types) and issubclass(valType, string_types):
             return
          elif issubclass(info.type, Link):
             # Allow object of expected type to be assigned to link
             if issubclass(valType, GetVmodlType(info.expectedType)):
                return
>         raise TypeError('For "%s" expected type %s, but got %s'
                        % (info.name, info.type.__name__, valType.__name__))
E         pyVmomi.SoapAdapter.ParserError: 'xml document: <http.client.HTTPResponse object at 0x10c43bfd0> parse error at: line:2, col:687'

../env/lib/python3.9/site-packages/pyVmomi/VmomiSupport.py:1097: ParserError
----------------------------------------------------- Captured log call ------------------------------------------------------
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.Folder:group-d1', 'selectSet': [{'name': 'visitFolders', 'type': 'vim.Folder', 'path': 'childEntity', 'selectSet': [{'name': 'visitFolders'}, {'name': 'dcToHf'}, {'name': 'dcToVmf'}, {'name': 'crToH'}, {'name': 'crToRp'}, {'name': 'HToVm'}, {'name': 'dsToVm'}, {'name': 'dcToDs'}]}, {'name': 'dcToVmf', 'type': 'vim.Datacenter', 'path': 'vmFolder', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'dcToHf', 'type': 'vim.Datacenter', 'path': 'hostFolder', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'crToH', 'type': 'vim.ComputeResource', 'path': 'host'}, {'name': 'crToRp', 'type': 'vim.ComputeResource', 'path': 'resourcePool', 'selectSet': [{'name': 'rpToRp'}, {'name': 'rpToVm'}]}, {'name': 'rpToRp', 'type': 'vim.ResourcePool', 'path': 'resourcePool', 'selectSet': [{'name': 'rpToRp'}, {'name': 'rpToVm'}]}, {'name': 'HToVm', 'type': 'vim.HostSystem', 'path': 'vm', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'rpToVm', 'type': 'vim.ResourcePool', 'path': 'vm'}, {'name': 'dsToVm', 'type': 'vim.Datastore', 'path': 'vm', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'dcToDs', 'type': 'vim.Datacenter', 'path': 'datastore', 'selectSet': [{'name': 'visitFolders'}]}]}]}], options={})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-57', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-60', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-63', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vim.VirtualMachine:vm-63' --> CreateSnapshot_Task(name=clean_snapshot, description=test clean snap for fubar host, memory=True, quiesce=True)
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vim.VirtualMachine:vm-63' <-- CreateSnapshot_Task
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> CreatePropertyCollector()
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- CreatePropertyCollector
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]a9f85d62-8b61-4f17-94a2-84a306216430' --> CreateFilter(spec={'propSet': [{'type': 'vim.Task', 'pathSet': ['info.state']}], 'objectSet': [{'obj': 'vim.Task:task-68'}]}, partialUpdates=True)
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]a9f85d62-8b61-4f17-94a2-84a306216430' <-- CreateFilter
DEBUG    infi.pyvmomi_wrapper.property_collector:property_collector.py:107 Checking for updates on property collector <TaskPropertyCollector: object_type=<class 'pyVmomi.VmomiSupport.vim.Task'>, properties=['info.state'], version=>
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]a9f85d62-8b61-4f17-94a2-84a306216430' --> WaitForUpdatesEx(version=, options={})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]a9f85d62-8b61-4f17-94a2-84a306216430' <-- WaitForUpdatesEx
DEBUG    infi.pyvmomi_wrapper.property_collector:property_collector.py:110 There is indeed a pending update
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]a9f85d62-8b61-4f17-94a2-84a306216430' --> DestroyPropertyCollector()
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]a9f85d62-8b61-4f17-94a2-84a306216430' <-- DestroyPropertyCollector
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.Task', 'all': False, 'pathSet': ['info']}], 'objectSet': [{'obj': 'vim.Task:task-68', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.Folder:group-d1', 'selectSet': [{'name': 'visitFolders', 'type': 'vim.Folder', 'path': 'childEntity', 'selectSet': [{'name': 'visitFolders'}, {'name': 'dcToHf'}, {'name': 'dcToVmf'}, {'name': 'crToH'}, {'name': 'crToRp'}, {'name': 'HToVm'}, {'name': 'dsToVm'}, {'name': 'dcToDs'}]}, {'name': 'dcToVmf', 'type': 'vim.Datacenter', 'path': 'vmFolder', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'dcToHf', 'type': 'vim.Datacenter', 'path': 'hostFolder', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'crToH', 'type': 'vim.ComputeResource', 'path': 'host'}, {'name': 'crToRp', 'type': 'vim.ComputeResource', 'path': 'resourcePool', 'selectSet': [{'name': 'rpToRp'}, {'name': 'rpToVm'}]}, {'name': 'rpToRp', 'type': 'vim.ResourcePool', 'path': 'resourcePool', 'selectSet': [{'name': 'rpToRp'}, {'name': 'rpToVm'}]}, {'name': 'HToVm', 'type': 'vim.HostSystem', 'path': 'vm', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'rpToVm', 'type': 'vim.ResourcePool', 'path': 'vm'}, {'name': 'dsToVm', 'type': 'vim.Datastore', 'path': 'vm', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'dcToDs', 'type': 'vim.Datacenter', 'path': 'datastore', 'selectSet': [{'name': 'visitFolders'}]}]}]}], options={})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-57', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['snapshot']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-57', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vim.VirtualMachine:vm-57' --> CreateSnapshot_Task(name=clean_snapshot, description=Clean backup snapshot, memory=False, quiesce=True)
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vim.VirtualMachine:vm-57' <-- CreateSnapshot_Task
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> CreatePropertyCollector()
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- CreatePropertyCollector
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]39f3094b-79cb-44da-92a1-e59cef53f602' --> CreateFilter(spec={'propSet': [{'type': 'vim.Task', 'pathSet': ['info.state']}], 'objectSet': [{'obj': 'vim.Task:task-70'}]}, partialUpdates=True)
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]39f3094b-79cb-44da-92a1-e59cef53f602' <-- CreateFilter
DEBUG    infi.pyvmomi_wrapper.property_collector:property_collector.py:107 Checking for updates on property collector <TaskPropertyCollector: object_type=<class 'pyVmomi.VmomiSupport.vim.Task'>, properties=['info.state'], version=>
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]39f3094b-79cb-44da-92a1-e59cef53f602' --> WaitForUpdatesEx(version=, options={})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]39f3094b-79cb-44da-92a1-e59cef53f602' <-- WaitForUpdatesEx
DEBUG    infi.pyvmomi_wrapper.property_collector:property_collector.py:110 There is indeed a pending update
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]39f3094b-79cb-44da-92a1-e59cef53f602' --> DestroyPropertyCollector()
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]39f3094b-79cb-44da-92a1-e59cef53f602' <-- DestroyPropertyCollector
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.Task', 'all': False, 'pathSet': ['info']}], 'objectSet': [{'obj': 'vim.Task:task-70', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.Folder:group-d1', 'selectSet': [{'name': 'visitFolders', 'type': 'vim.Folder', 'path': 'childEntity', 'selectSet': [{'name': 'visitFolders'}, {'name': 'dcToHf'}, {'name': 'dcToVmf'}, {'name': 'crToH'}, {'name': 'crToRp'}, {'name': 'HToVm'}, {'name': 'dsToVm'}, {'name': 'dcToDs'}]}, {'name': 'dcToVmf', 'type': 'vim.Datacenter', 'path': 'vmFolder', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'dcToHf', 'type': 'vim.Datacenter', 'path': 'hostFolder', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'crToH', 'type': 'vim.ComputeResource', 'path': 'host'}, {'name': 'crToRp', 'type': 'vim.ComputeResource', 'path': 'resourcePool', 'selectSet': [{'name': 'rpToRp'}, {'name': 'rpToVm'}]}, {'name': 'rpToRp', 'type': 'vim.ResourcePool', 'path': 'resourcePool', 'selectSet': [{'name': 'rpToRp'}, {'name': 'rpToVm'}]}, {'name': 'HToVm', 'type': 'vim.HostSystem', 'path': 'vm', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'rpToVm', 'type': 'vim.ResourcePool', 'path': 'vm'}, {'name': 'dsToVm', 'type': 'vim.Datastore', 'path': 'vm', 'selectSet': [{'name': 'visitFolders'}]}, {'name': 'dcToDs', 'type': 'vim.Datacenter', 'path': 'datastore', 'selectSet': [{'name': 'visitFolders'}]}]}]}], options={})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-57', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['name']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-60', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['snapshot']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-60', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vim.VirtualMachine:vm-60' --> CreateSnapshot_Task(name=clean_snapshot, description=Clean backup snapshot, memory=False, quiesce=True)
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vim.VirtualMachine:vm-60' <-- CreateSnapshot_Task
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> CreatePropertyCollector()
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- CreatePropertyCollector
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]98f9f539-249b-4252-a037-3c5bf8e084a4' --> CreateFilter(spec={'propSet': [{'type': 'vim.Task', 'pathSet': ['info.state']}], 'objectSet': [{'obj': 'vim.Task:task-72'}]}, partialUpdates=True)
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]98f9f539-249b-4252-a037-3c5bf8e084a4' <-- CreateFilter
DEBUG    infi.pyvmomi_wrapper.property_collector:property_collector.py:107 Checking for updates on property collector <TaskPropertyCollector: object_type=<class 'pyVmomi.VmomiSupport.vim.Task'>, properties=['info.state'], version=>
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]98f9f539-249b-4252-a037-3c5bf8e084a4' --> WaitForUpdatesEx(version=, options={})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]98f9f539-249b-4252-a037-3c5bf8e084a4' <-- WaitForUpdatesEx
DEBUG    infi.pyvmomi_wrapper.property_collector:property_collector.py:110 There is indeed a pending update
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]98f9f539-249b-4252-a037-3c5bf8e084a4' --> DestroyPropertyCollector()
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]98f9f539-249b-4252-a037-3c5bf8e084a4' <-- DestroyPropertyCollector
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.Task', 'all': False, 'pathSet': ['info']}], 'objectSet': [{'obj': 'vim.Task:task-72', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.VirtualMachine', 'all': False, 'pathSet': ['snapshot']}], 'objectSet': [{'obj': 'vim.VirtualMachine:vm-63', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vim.vm.Snapshot:snapshot-69' --> RevertToSnapshot_Task(host=None, suppressPowerOn=False)
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vim.vm.Snapshot:snapshot-69' <-- RevertToSnapshot_Task
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> CreatePropertyCollector()
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- CreatePropertyCollector
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]65e77307-660d-437d-ae8f-185c82802fb4' --> CreateFilter(spec={'propSet': [{'type': 'vim.Task', 'pathSet': ['info.state']}], 'objectSet': [{'obj': 'vim.Task:task-74'}]}, partialUpdates=True)
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]65e77307-660d-437d-ae8f-185c82802fb4' <-- CreateFilter
DEBUG    infi.pyvmomi_wrapper.property_collector:property_collector.py:107 Checking for updates on property collector <TaskPropertyCollector: object_type=<class 'pyVmomi.VmomiSupport.vim.Task'>, properties=['info.state'], version=>
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]65e77307-660d-437d-ae8f-185c82802fb4' --> WaitForUpdatesEx(version=, options={'maxWaitSeconds': 60})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]65e77307-660d-437d-ae8f-185c82802fb4' <-- WaitForUpdatesEx
DEBUG    infi.pyvmomi_wrapper.property_collector:property_collector.py:110 There is indeed a pending update
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]65e77307-660d-437d-ae8f-185c82802fb4' --> DestroyPropertyCollector()
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:session[178b4705-b275-4877-84d6-a33b41ed007a]65e77307-660d-437d-ae8f-185c82802fb4' <-- DestroyPropertyCollector
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' --> RetrievePropertiesEx(specSet=[{'propSet': [{'type': 'vim.Task', 'all': False, 'pathSet': ['info']}], 'objectSet': [{'obj': 'vim.Task:task-74', 'skip': False}]}], options={'maxObjects': 1})
DEBUG    infi.pyvmomi_wrapper.connect:connect.py:20 'vmodl.query.PropertyCollector:propertyCollector' <-- RetrievePropertiesEx
================================================== short test summary info ===================================================
FAILED tests/test_vmware_utils.py::test_enable_clean_snapshots - pyVmomi.SoapAdapter.ParserError: 'xml document: <http.clie...
```
Am I doing something wrong here? I tried issuing `vcsim -trace` and it keeps waiting after this (trimmed log except for the last lines):
```
[2021-02-11 16:04:08.829943] INFO: infradev_cli.vmware_utils: Reverting to snapshot clean_snapshot
[2021-02-11 16:04:08.829995] DEBUG: infradev_cli.vmware_utils: Waiting for 60 seconds for tasks: ('vim.Task:task-74',)
[2021-02-11 16:04:08.830134] DEBUG: infi.pyvmomi_wrapper.connect: 'vmodl.query.PropertyCollector:propertyCollector' --> CreatePropertyCollector()
[2021-02-11 16:04:08.830713] DEBUG: infi.pyvmomi_wrapper.connect: 'vmodl.query.PropertyCollector:propertyCollector' <-- CreatePropertyCollector
[2021-02-11 16:04:08.830950] DEBUG: infi.pyvmomi_wrapper.connect: 'vmodl.query.PropertyCollector:session[e3daf85f-2b3e-4ba2-8c44-f07a63a49de8]dd150fbf-f05e-4b8a-ba1b-ba55048177aa' --> CreateFilter(spec={'propSet': [{'type': 'vim.Task', 'pathSet': ['info.state']}], 'objectSet': [{'obj': 'vim.Task:task-74'}]}, partialUpdates=True)
```

--- a/govc/test/snapshot.bats
+++ b/govc/test/snapshot.bats
@@ -75,6 +75,11 @@ test_vm_snapshot() {
 
   # set current to grand
   run govc snapshot.revert -vm "$vm" grand
+  assert_success
+
+  vm_id=$(govc find -i vm -name "$vm")
+  entity=$(govc object.collect -s TaskManager:TaskManager recentTask | awk -F, '{print $NF}' | xargs -I% govc object.collect -s % info.entity)
+  assert_equal "$vm_id" "$entity"
 
   # name is unique
   run govc snapshot.remove -vm "$vm" grand

--- a/simulator/snapshot.go
+++ b/simulator/snapshot.go
@@ -147,7 +147,7 @@ func (v *VirtualMachineSnapshot) RemoveSnapshotTask(ctx *Context, req *types.Rem
 }
 
 func (v *VirtualMachineSnapshot) RevertToSnapshotTask(req *types.RevertToSnapshot_Task) soap.HasFault {
-	task := CreateTask(v, "revertToSnapshot", func(t *Task) (types.AnyType, types.BaseMethodFault) {
+	task := CreateTask(v.Vm, "revertToSnapshot", func(t *Task) (types.AnyType, types.BaseMethodFault) {
 		vm := Map.Get(v.Vm).(*VirtualMachine)
 
 		Map.WithLock(vm, func() {

